### PR TITLE
8296243: [IR Framework] Fix issues with IRNode.ALLOC* regexes

### PR DESCRIPTION
**Pull request by chhagedorn**
There are currently two problems with `IRNode.ALLOC*` regexes:
1. On PPC64, we do not account for an `LI` instruction which matches the array size. As a result, we could miss some array allocations with the `ALLOC_ARRAY*` regexes:
```
2e4     LD      R3, offset, R3   // load ptr precise [java/lang/Object:
0x0000200058006e40 *: :Constant:exact * from TOC (lo)
2e8     STD     R17, [R1_SP + #104+0]    // spill copy
2ec     LI      R4, #1    <------- we only look for LGHI here which is specific to s390 while LI is used for PPC64
2f0     CALL,static 0x00002000177cd300   // ==>  wrapper for: _new_array_Java
```
This was revealed by a new test added by [JDK-8280378](https://bugs.openjdk.org/browse/JDK-8280378) but was already a problem before this change.

2. The newly added `IRNode.ALLOC*` regexes in JDK-8280378 which can be matched on the independent ideal compile phases by using the name of the IR node "Allocate" also matches "AllocateArray" (substring match). This is unexpected. I've changed this by matching "Allocate" exactly.

I've additionally removed the matching of `LI` and `LGHI` for the `ALLOC` regexes on normal objects as we do not have an array size. I think it's safe to remove these (might need some additional testing on PPC64/s390).

Thanks @TheRealMDoerr for helping to test the initial fix on PPC64!

Thanks,
Christian

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8296243](https://bugs.openjdk.org/browse/JDK-8296243): [IR Framework] Fix issues with IRNode.ALLOC* regexes

### Reviewers
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**) ⚠️ Review applies to [c36e645b](https://git.openjdk.org/jdk/pull/11037/files/c36e645b6c0b62f5ef01b52e3a7a56383ae4d3a7)
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**) ⚠️ Review applies to [0d28220f](https://git.openjdk.org/jdk/pull/11037/files/0d28220fe507a8f261fd7a07916b0added30025c)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11037/head:pull/11037` \
`$ git checkout pull/11037`

Update a local copy of the PR: \
`$ git checkout pull/11037` \
`$ git pull https://git.openjdk.org/jdk pull/11037/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11037`

View PR using the GUI difftool: \
`$ git pr show -t 11037`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11037.diff">https://git.openjdk.org/jdk/pull/11037.diff</a>

</details>

### CODE REVIEW

**Review by TheRealMDoerr — APPROVED** [review]
LGTM, but I'm not very familiar with the IR framework. The test `TestPhaseIRMatching.java` has passed on PPC64 and s390.

**Comment by chhagedorn** [comment]
Thanks Martin for your review and testing it again!

File: test/hotspot/jtreg/testlibrary_tests/ir_framework/tests/TestPhaseIRMatching.java [hunk 1]
```diff
@@ -243,6 +245,15 @@ public void alloc() {
         obj = new Object();
         obj2 = new Object[1];
     }
+    @Test
```
Comment (TobiHartmann) [inline test/hotspot/jtreg/testlibrary_tests/ir_framework/tests/TestPhaseIRMatching.java:248]: I think a newline before `@Test` would be good.

**Review by TobiHartmann — APPROVED** [review]
Looks good to me.

**Comment by chhagedorn** [comment]
Thanks Tobias for your review! I've added a new line as suggested.

**Review by TobiHartmann — APPROVED** [review]

**Pull request closed without merging**